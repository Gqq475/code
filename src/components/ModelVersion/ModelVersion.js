import React from 'react'
import classes from './ModelVersion.scss'
import ModelList from 'components/ModelList'
import ModelDetail from 'components/ModelDetail'
type Props = {
  data: Object,
};
export class ModelVersion extends React.Component {
  props: Props;
  constructor (props) {
    super(props)
    this.state = {
      detailData: {}
    }
    this.setDetail = this.setDetail.bind(this)
  }

  setDetail (data) {
    this.setState({
      detailData: data
    })
  }

  componentWillReceiveProps (nextProps) {
    this.setState({
      detailData: {}
    })
  }

  render () {
    return (
      <div className={classes['ModelVersion-container']}>
        <ModelList
          data={this.props.data}
          setDetail={this.setDetail}
        />
        {
          <ModelDetail
            data={this.state.detailData} />
        }
      </div>
    )
  }
}

export default ModelVersion
