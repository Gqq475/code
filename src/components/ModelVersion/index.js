import ModelVersion from './ModelVersion'
export default ModelVersion
