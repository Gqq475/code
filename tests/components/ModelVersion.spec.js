import React from 'react'
import ModelVersion from 'components/ModelVersion/ModelVersion'

describe('(Component) ModelVersion', () => {
  it('render normally', () => {
    const wrapper = shallow(
      <ModelVersion />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
