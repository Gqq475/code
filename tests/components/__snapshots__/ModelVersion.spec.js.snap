// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) ModelVersion render normally 1`] = `
<div
  className="ModelVersion-container"
>
  <Connect(ModelList)
    setDetail={[Function]}
  />
  <ModelDetail
    data={Object {}}
  />
</div>
`;
